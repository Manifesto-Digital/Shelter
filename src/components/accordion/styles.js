import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
