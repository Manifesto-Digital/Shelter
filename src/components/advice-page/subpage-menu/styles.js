import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
