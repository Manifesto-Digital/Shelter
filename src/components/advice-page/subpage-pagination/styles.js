import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
