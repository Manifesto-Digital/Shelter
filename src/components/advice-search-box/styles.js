import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
