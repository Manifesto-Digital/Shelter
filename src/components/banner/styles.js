import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
