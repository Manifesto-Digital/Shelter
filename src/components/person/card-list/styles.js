import styled, { css } from 'styled-components';
import { buttonReset } from '../../styled/buttons';
import { breakpoint } from '../../theme/breakpoint';

export const Wrapper = styled.section`
  margin-bottom: ${({ theme }) => theme.spacing.large};
  ${({ borderTop }) =>
    borderTop &&
    css`
      margin-top: ${({ theme }) => theme.spacing.large};
      border-top: 1px solid ${({ theme }) => theme.palette.grey10};
      padding-top: ${({ theme }) => theme.spacing.standard};
    `};
`;

export const List = styled.ul`
  padding: 0;
  display: grid;

  ${breakpoint.desktop`
    grid-template-columns: ${({ columns }) =>
      columns && `repeat(${columns}, 1fr)`};
    grid-gap: ${({ theme }) => theme.spacing.standard};
  `};
`;

export const LoadMore = styled.div`
  text-align: center;
  visibility: ${({ visible }) => (visible ? 'visible' : 'hidden')};
`;

export const LoadMoreButton = styled.button`
  ${buttonReset}
  color: ${({ theme }) => theme.palette.primary};
  padding: ${({ theme }) => theme.spacing.small};
  
  svg { 
    width: 35px;
    height: 35px;
  }
`;
