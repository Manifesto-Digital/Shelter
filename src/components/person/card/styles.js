import SVG from 'react-inlinesvg';
import styled from 'styled-components';
import { breakpoint } from '../../theme/breakpoint';
import Image from '../../image';

export const Card = styled.div`
  display: flex;
  flex-wrap: wrap;
  box-shadow: ${({ theme }) => theme.boxshadow.small};
  position: relative;
  margin-bottom: ${({ theme }) => theme.spacing.standard};
  width: 100%;

  &:hover {
    box-shadow: ${({ hasLink, theme }) => hasLink && theme.boxshadow.standard};
  }

  ${breakpoint.desktop`
    width: ${({ theme, columns }) =>
      columns === 2 && `calc(50% - ${theme.spacing.small})`};

    font-size: ${({ theme, columns }) => columns && theme.fontsize.small};
  `};
`;

export const Info = styled.div`
  padding: ${({ theme }) => `${theme.spacing.small} ${theme.spacing.standard}`};
  flex: 1;
`;

export const Name = styled.h3`
  margin-bottom: 5px;
`;

export const JobTitle = styled.p`
  margin-bottom: ${({ theme }) => theme.spacing.small};
  font-weight: ${({ theme }) => theme.fontWeight.bold};
  line-height: 1.2;
`;

export const StyledImage = styled(Image)`
  align-self: center;
  margin-bottom: ${({ theme }) => theme.spacing.small};
  width: 100%;

  ${breakpoint.mobileLand`
    width: 35%;
    margin-bottom: 0;
  `};
`;

export const ContactLine = styled.p`
  margin-bottom: 5px;

  &:last-of-type {
    margin-bottom: 0;
  }
`;

export const ContactIcon = styled(SVG)`
  margin-right: ${({ theme }) => theme.spacing.small};

  svg {
    width: 18px;
    height: 18px;
  }
  path {
    fill: ${({ theme }) => theme.palette.primary};
  }
`;
