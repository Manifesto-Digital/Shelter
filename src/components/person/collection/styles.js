import styled from 'styled-components';

export const Wrapper = styled.section`
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-top: ${({ theme }) => theme.spacing.standard};
`;

export const Header = styled.h2`
  margin-bottom: ${({ theme }) => theme.spacing.large};
`;
