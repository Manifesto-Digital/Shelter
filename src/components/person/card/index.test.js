import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'gatsby';
import { snapshotComponent, mountWithTheme } from 'test-helpers';
import {
  createFactory,
  createPerson,
  createInternalLink,
} from '../../../utils/test-factories';
import PersonCard from '.';
import { Card } from './styles';
import { hidePascalCaseWarning } from '../../../utils/test-mocks';
import { sizes, emSize } from '../../theme/breakpoint';

const createCardProps = createFactory({
  person: createPerson(),
  link: createInternalLink(),
});

it('renders correctly', () => {
  const mockData = createCardProps();
  snapshotComponent(<PersonCard {...mockData} />);
});

it('renders the persons name and job title', () => {
  const mockData = createCardProps({
    person: {
      firstName: 'first name mock',
      lastName: 'last name moc',
      jobTitle: 'mock job title',
    },
  });
  const wrapper = shallow(<PersonCard {...mockData} />);

  expect(wrapper.text()).toContain(`${mockData.person.firstName}`);
  expect(wrapper.text()).toContain(`${mockData.person.lastName}`);
  expect(wrapper.text()).toContain(`${mockData.person.jobTitle}`);
});

it('renders the persons image', () => {
  const personProps = createPerson({
    photo: {
      description: 'mock description',
      file: {
        url: 'http://mockurl.com',
      },
    },
  });
  const wrapper = mountWithTheme(<PersonCard person={personProps} />);
  expect(wrapper.find('img').prop('src')).toContain(personProps.photo.file.url);
});
hidePascalCaseWarning();

it('renders a link if passed', () => {
  const mockData = createCardProps({
    link: {
      slug: '/mock-page',
    },
  });
  const wrapper = mountWithTheme(<PersonCard {...mockData} />);
  expect(wrapper.find(Link)).toHaveLength(1);
});

it('has 50% width on desktop if 2 columns specified', () => {
  const mockData = createCardProps({
    columns: 2,
  });
  const wrapper = mountWithTheme(<PersonCard {...mockData} />);
  expect(wrapper.find(Card)).toHaveStyleRule(
    'width',
    expect.stringContaining('50%'), // Allow for any gutter size in calc
    {
      media: `(min-width: ${emSize(sizes.desktop)})`,
    }
  );
});
