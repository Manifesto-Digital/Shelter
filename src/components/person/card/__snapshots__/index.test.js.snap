// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c9 {
  padding: 0;
  margin: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  -webkit-clip-path: inset(1px);
  clip-path: inset(1px);
  width: 1px;
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
}

.c7 {
  border-bottom-right-radius: 3px;
  background: #8c8c8c;
  color: #fff;
  padding-top: 3px;
  height: 30px;
  width: 30px;
  position: absolute;
  right: 3px;
  bottom: 3px;
}

.c7 svg {
  width: 22px;
  height: 22px;
  display: block;
  margin: 0 auto;
}

.c8 {
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-shadow: 0 1.5px 4px rgba(0,0,0,0.12),0 1.5px 6px rgba(0,0,0,0.06);
  position: relative;
  margin-bottom: 20px;
  width: 100%;
}

.c0:hover {
  box-shadow: 0 2.5px 6px rgba(0,0,0,0.16),0 2.5px 6px rgba(0,0,0,0.13);
}

.c2 {
  padding: 10px 20px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  margin-bottom: 5px;
}

.c4 {
  margin-bottom: 10px;
  font-weight: 500;
  line-height: 1.2;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-bottom: 10px;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  word-break: break-word;
  margin-bottom: 10px;
  line-height: 1;
}

.c5:last-of-type {
  margin-bottom: 0;
}

.c6 {
  margin-right: 10px;
}

.c6 svg {
  width: 18px;
  height: 18px;
}

.c6 path {
  fill: #ea232d;
}

@media (min-width:32em) {
  .c2 {
    width: 70%;
    margin-bottom: 0;
  }
}

@media (min-width:32em) {
  .c1 {
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
    margin-top: 10px;
    width: 30%;
    margin-bottom: 0;
  }
}

<div
  className="c0"
>
  <img
    alt="Collection hero"
    className="c1"
    src="//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?fit=fill&h=500&w=500"
  />
  <div
    className="c2"
  >
    <h3
      className="c3"
    >
      Test
       
      Person
    </h3>
    <p
      className="c4"
    >
      Developer
    </p>
    <p
      className="c5"
    >
      <span
        className="isvg loading c6"
      />
       
      test@test.com
    </p>
    <p
      className="c5"
    >
      <span
        className="isvg loading c6"
      />
       
      012345678910
    </p>
  </div>
  <div
    aria-hidden="true"
    className="c7"
  >
    <span
      className="isvg loading"
    />
  </div>
  <Link
    className="tracking-card c8"
    data-tracking="people-collection"
    to="/internal-link-slug"
  >
    <span
      className="c9"
    >
      Test
       
      Person
    </span>
  </Link>
</div>
`;
